core/src/main/java/com/datastax/oss/driver/api/core/DefaultProtocolVersion.java

```diff
@@ -30,14 +30,17 @@
   /** Version 4, supported by Cassandra 2.2 and above. */
   V4(ProtocolConstants.Version.V4, false),
 
+  /** Version 5, supported by Cassandra 4.0 and above. */
+  V5(ProtocolConstants.Version.V5, false),
+
   /**
-   * Version 5, currently supported as a beta preview in Cassandra 3.10 and above.
+   * Version 6, currently supported as a beta preview in Cassandra 4.0 and above.
    *
    * <p>Do not use this in production.
    *
    * @see ProtocolVersion#isBeta()
    */
-  V5(ProtocolConstants.Version.V5, true),
+  V6(ProtocolConstants.Version.V5, true),
```

Comment:
`ProtocolConstants.Version` should be `V6`, you will probably have to add it to `native-protocol`.

Reply:
added in https://github.com/datastax/native-protocol/pull/35